### ISSUE TITLE
Multiple GPU's across network (Distributed Inference)

### ISSUE THREAD

**Issue by RamishSiddiqui** [oobabooga/text-generation-webui#5600]
**Description**

Supporting distributed inference using multiple GPU on the local or on cloud.

**Comment by aikitoria** [comment]
This would be incredibly slow, multi-GPU inference requires very low latency and high bandwidth for the connection between them.

**Comment by RamishSiddiqui** [comment]
Agreed, but some users of this repo might not have state of the art GPU not to mention multiple high end GPU's on a single machine. Also they could have private on prem cloud. In my case i have on-prem private cloud with multiple server with multiple GPU which i use for distributed training using horovod/mpi or pytorch's distributed computing. I would like to do the same for this too.

**Comment by RamishSiddiqui** [comment]
I do think on prem servers wouldn't have the issue of bandwidth, but remote inference to cloud servers(AWS, Azure, GCP) might provide a bottle but this too entirely depends on your internet connection speed.

One way to also overcome this thing would be too as @oldmanjk mentioned keeping copies of model on remote nodes and only passing data for inference and the respective instruction set i.e which layers to load on the GPU and do a forward pass using the data.

**Comment by aikitoria** [comment]
Something that does work very well in practice is running inference on a cloud server with multiple GPUs all in the same machine.

**Comment by Tedy50** [comment]
This woud be absolutely useless idea because even running those GPUs on your local machine already slows down inference significantly  the problem is that you can't run those GPUs in parallel unless it is mixtral type model where you basically run multiple models and then average their output.

**Comment by RamishSiddiqui** [comment]
It seems like vLLM is already utilizing this technique to run bigger LLMs on consumer hardware.
They are achieving this utilizing Ray library.
It would've been nice to see text-gen ui utilizing this technique too.

You can read it here for more info: https://docs.vllm.ai/en/latest/serving/distributed_serving.html
